src/main/java/quizapp/domain/service/favorite/FavoriteService.java

```diff
@@ -0,0 +1,62 @@
+package quizapp.domain.service.favorite;
+
+import java.io.Serializable;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import quizapp.domain.model.Favorite;
+import quizapp.domain.model.Quiz;
+import quizapp.domain.model.User;
+import quizapp.domain.repository.favorite.FavoriteRepository;
+import quizapp.domain.service.user.QuizUserDetails;
+
+@Service
+@Transactional
+public class FavoriteService implements Serializable {
+  @Autowired
+  FavoriteRepository favoriteRepository;
+
+  public Favorite addFavorite(User user, Quiz quiz) {
+    boolean overlap = favoriteRepository.findByUserAndQuiz(user, quiz) != null;
+    if (overlap) {
+      throw new AlreadyAddedFavoriteException("重複予約");
+    }
+    Favorite favorite = new Favorite();
+    ZonedDateTime now = ZonedDateTime.now(ZoneId.of("Asia/Tokyo"));
+    favorite.setUser(user);
+    favorite.setQuiz(quiz);
+    favorite.setCreatedAt(now);
+    favoriteRepository.save(favorite);
+    return favorite;
+  }
+
+  public void removeFavorite(User user, Quiz quiz) {
+    Favorite favorite = favoriteRepository.findByUserAndQuiz(user, quiz);
+    favoriteRepository.delete(favorite);
+  }
+
+  public List<Quiz> findFavorites(QuizUserDetails quizUserDetails) {
+    List<Favorite> favs =
+        favoriteRepository.findByUserIdOrderByCreatedAtAsc(quizUserDetails.getUser().getId());
+    List<Quiz> favorites = new ArrayList<Quiz>();
```

Comment:
```suggestion
    List<Quiz> favoriteQuizzes = new ArrayList<Quiz>();
```

favsと混同するかつ `List<Quiz>` なのでこちらの方が適切かと思いましたが、お任せします。

Reply:
ありがとうございます、命名は一度見直します。

Reply:
> favsと混同するかつ List<Quiz> なのでこちらの方が適切かと思いましたが、お任せします。
ありがとうございます、命名は一度見直します。